src/runtime/composables/search.ts

```diff
@@ -0,0 +1,113 @@
+import MiniSearch, { type Options as MiniSearchOptions } from 'minisearch'
+import type { UseFetchOptions } from 'nuxt/app'
+import { useRuntimeConfig, useFetch } from '#imports'
+
+export const defineMiniSearchOptions = <DataItem>(options: MiniSearchOptions<DataItem>) => {
+  return ref(options)
+}
+
+export const searchContent = async <DataItem>(search: MaybeRefOrGetter<string>, options: { miniSearch?: MaybeRefOrGetter<MiniSearchOptions<DataItem>>, fetch?: MaybeRefOrGetter<UseFetchOptions<string | DataItem[]>> } = {}) => {
+  const runtimeConfig = useRuntimeConfig()
+  const { content } = runtimeConfig.public
+  const { integrity, api: { baseURL: baseAPI }, search: searchOptions } = content
+  const { indexed: useIndexedSearch } = searchOptions || {}
+
+  if (useIndexedSearch) {
+    const { options: miniSearchOptions } = searchOptions || {}
+
+    const { data } = await useFetch<string>(`${baseAPI}/indexed-search${integrity ? '-' + integrity : ''}`, { responseType: 'text', ...options.fetch as MaybeRefOrGetter<UseFetchOptions<string>> | undefined })
+
+    // We need for a computed since user can have a lazy fetch.
+    return computed(() => {
+      if (!data.value) {
+        return '[]'
+      }
+
+      const { results } = useIndexedMiniSearch(search, data as Ref<string>, miniSearchOptions!)
+
+      return results.value
+    })
+  }
```

Comment:
There are two watchers and one computed inside`useIndexedMiniSearch`, and here we wrapped this function call inside a computed. It seems not right.
I think using `$fetch` and letting user call `searchContent` inside `useAysncData` would be an approach for supporting lazy search.

Reply:
I'll remove it and I think that users usage will tell use!